internal/juju/applications.go

```diff
@@ -1413,6 +1413,11 @@ func (c applicationsClient) computeSetCharmConfig(
 	// is called.
 	if input.Revision != nil {
 		oldOrigin.Revision = input.Revision
+		// This code is coupled with the previous if input.Revision != nil.
+		// The idea is that deploying with ID and Hash set to "" will force Juju to find the revision set by the user.
+		// More info in the previous if input.Revision != nil comment.
```

Comment:
This is good, minor tweak and you can probably drop the last sentence.
```suggestion
		// This code is coupled with the previous `if input.Revision != nil`.
		// The idea is that deploying with ID and Hash set to "" will force Juju to find the revision set by the user.
```